packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -1368,7 +1368,34 @@ class FocusManager with DiagnosticableTreeMixin, ChangeNotifier {
   /// the [WidgetsBinding] instance.
   static FocusManager get instance => WidgetsBinding.instance.focusManager;
 
-  bool _lastInteractionWasTouch = true;
+  bool get _lastInteractionWasTouch {
+    // Assume that if we're on one of these mobile platforms, or if there's no
+    // mouse connected, that the initial interaction will be touch-based, and
```

Comment:
Where are we checking if there's a mouse connected?

Reply:
Good point. I had removed it here because in the absence of any events, there's no way to know if the mouse is actually connected until an event occurs.  I think I'll add it back in for mobile, though, because by default (before any events), `mouseIsConnected` returns false, but if an event occurs before _lastInteractionWasTouch is accessed it could improve accuracy.

I don't want to make the default for desktop platforms depend on `mouseIsConnected`, however, because then the default will almost always start off as touch, which is probably wrong, and the moment it gets any pointer event it will be updated to be correct anyhow.

Mainly, this change is just trying to make a good educated guess before there's any information available, and for most apps, it's kind of academic: as soon as the user interacts, it will update to the right value anyhow.